plugins/techdocs-backend/src/service/router.ts

```diff
@@ -226,13 +231,35 @@ export async function createRouter(
     );
   });
 
+  // Route treated similarly to /static/docs
+  // Requires an existing entity in the catalog for authorization
+  router.use(
+    '/static/entity/docs/:namespace/:kind/:name',
+    async (req, _res, next) => {
+      const { kind, namespace, name } = req.params;
+      const entityName = { kind, namespace, name };
+      const token = getBearerToken(req.headers.authorization);
+
+      const entity = await catalogClient.getEntityByName(entityName, { token });
```

Comment:
Y'all are probably already thinking about this, but just to highlight: making a call to the catalog backend for each asset/object loaded is going to be heavy.  ...Both in terms of throughput on the catalog backend, but also latency on the TechDocs experience (especially when they aren't colocated).

For example [this page on the demo site](https://demo.backstage.io/docs/default/component/backstage/features/software-catalog/) would result in 8 very rapid calls back to the Catalog client for the same resource.

Reply:
We're considering a very low ttl cache for the auth responses. @Rugvip also suggested generating tokens or signed urls for these assets and authorizing against that instead